src/routes/pages.ts

```diff
@@ -189,13 +189,22 @@ function setRoutes(redisClient: redis.RedisClient): express.Router {
         specificGenres.sort();
         days.sort();
 
+        let dates: Date[]     = await redisHelper.getLineupDates(redisClient, festival.name, queryYear);
         const lastUpdatedDate = await redisHelper.getLineupLastUpdatedDate(redisClient, festival.name, queryYear);
 
+        // if calendar dates are found, much match exactly to date index
+        if (dates.length > 0 && (daysMap.size !== dates.length)) {
+            console.error(`Mismatch between number of days (${daysMap.size}) and dates (${dates.length}): ${
+                festival.name} ${queryYear}`);
+            dates = [];
+        }
+
         res.render("customize-list", {
             prod : process.env.DEPLOY_STAGE === 'PROD',
             titleOverride : `Customize Playlist - ${festival.display_name} ${queryYear}`,
             festival,
             festivalYear : queryYear,
+            dates,
```

Comment:
I started looking at this and it's a bit fiddly, because currently `daysMap` is actually built up through iterating through the artist list, and the days are essentially extracted from there, whereas now we would be going to Redis to get the days and using that as the canonical list. Any thoughts on how you want that to work? I'm thinking probably get the days+metadata from redis before the `for (const artist of artists)` loop, but then need to look at all the checkbox logic too.

Reply:
Nevermind the above, will push some proposed changes.